### PR TITLE
add three charts

### PR DESCRIPTION
**Pull request by JiaoJiao123**
Three new charts
1: number of samples per patient
2: with mutation data
3: with cna data

Added features:
1: hide the chart if there is only one value 
2: decide chart type(pie chart or bar chart) only by data type

### CODE REVIEW

**Comment by JiaoJiao123** [comment]
Related to pull request #1428

**Comment by JiaoJiao123** [comment]
@zhx828  Could you help with the code review? Thank you

**Comment by JiaoJiao123** [comment]
bug:  cancer_study_id=brca_metabric still show 'With CNA Data' pie chart even though it only has one value. Same with study brca_tcga_pub2015 'With Mutation Data'

@zhx828 That's a good catch. Thank you so much hongxin! I fixed it and if you could check it out that would be great. I am sorry about the inconvenience of switching to a new pr.

File: portal/src/main/webapp/js/src/study-view/data/StudyViewProxy.js [hunk 1]
```diff
@@ -362,55 +362,57 @@ var StudyViewProxy = (function() {
...
+                sample_count_patient_caseAttr.attr_id = 'SAMPLE_COUNT_PATIENT';
+                sample_count_patient_caseAttr.display_name = '# of Samples Per Patient';
+                sample_count_patient_caseAttr.description = '';
+                sample_count_patient_caseAttr.datatype = 'STRING';
+                sample_count_patient_caseAttr.numOfNoneEmpty =  obtainDataObject.arr.length;
+
+                var currentItem, uniqueCounts = [1], maxCount = 1, sequencedValues = [], hasCNAValues = [];
 
                 for(var i = 0; i < obtainDataObject.arr.length; i++){
                     currentItem = obtainDataObject.arr[i];
```
Comment (zhx828) [inline portal/src/main/webapp/js/src/study-view/data/StudyViewProxy.js]: Do you still need 'currentItem' ?

**Pull request merged**